### PR TITLE
Added information about  outerClickHandler option

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this pull request

This pull request (PR) adds information about `outerClickHandler` for the [Modal Widget](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html).

Additionally, the list of options was sorted.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html

## Links to Magento source code

-  https://github.com/magento/magento2/blob/2.3/app/code/Magento/Ui/view/base/web/js/modal/modal.js#L431
-  https://github.com/magento/magento2/blob/2.3/app/code/Magento/Ui/view/base/web/js/modal/modal.js#L443

whatsnew
Added information about the `outerClickHandler` option for the [Modal Widget](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.3/javascript-dev-guide/widgets/widget_modal.md [hunk 1]
```diff
@@ -218,6 +219,14 @@ The class that is assigned to parent when the modal is opened.
 
 **Default value**: `_has-modal`
 
+### `outerClickHandler` {#modal_outerClickHandler}
+
+The function that handles the click on the modal overlay. If `outerClickHandler` option is not specified, the [closeModal()](#modal_close) method is used for handling the click on the modal overlay.
```
Comment (rogyar) [inline src/guides/v2.3/javascript-dev-guide/widgets/widget_modal.md]: I'm not sure it's a great idea to use "click on the modal overlay" statement. The "modal overlay" definition might be ambiguous in this case. If you mean the area out of the modal window, it's not 100% overlay according to the classic definition. 
"overlay" usually means something that overlaps the background (so, the modal window itself in our case).

**Comment by serhiyzhovnir** [comment]
Hi @rogyar 
I've adjusted the sentence for `outerClickHandler` option description.
Please, review this PR again.
Thank you!

**Comment by dobooth** [comment]
running tests

**Pull request merged**